addons/mail/models/mail_message.py

```diff
@@ -944,12 +944,22 @@ def _notify_moderators(self):
     # MESSAGE READ / FETCH / FAILURE API
     # ------------------------------------------------------
 
-    def _message_format(self, fnames):
-        """Reads values from messages and formats them for the web client."""
+    def _message_format(self, fnames, fvalues_list=None):
+        """Reads values from messages and formats them for the web client.
+
+        :param fvalues_list: see ``MailMessage.message_format()``
+        """
+        if fvalues_list is None:
+            fvalues_list = [dict()] * len(self)
+            read_fnames = fnames
+        else:
+            read_fnames = [fname for fname in fnames if fname not in fvalues_list[0]]
+
         self.check_access_rule('read')
-        vals_list = self._read_format(fnames)
+        vals_list = self._read_format(read_fnames)
         safari = request and request.httprequest.user_agent.browser == 'safari'
-        for vals in vals_list:
+        for vals, fvalues in zip(vals_list, fvalues_list):
+            vals.update(**fvalues)
```

Comment:
Are you sure about the need for this change?
I think the values already fetched should be correctly read from cache instead of making query when using `_read_format`. It was actually made for that specific purpose.

Reply:
Maybe it was done on that purpose ... but it does not work as there are extra queries ^^ . When there is no-one to notify we loose queries here.